recorder.go

```diff
@@ -178,8 +190,8 @@ func NewRecorder(opts Options) basictracer.SpanRecorder {
 	if _, found := opts.Tags[ComponentNameKey]; !found {
 		opts.Tags[ComponentNameKey] = path.Base(os.Args[0])
 	}
-	if _, found := opts.Tags[GUIDKey]; !found {
-		opts.Tags[GUIDKey] = genSeededGUID()
+	if _, found := opts.Tags[GUIDKey]; found {
+		panic(fmt.Sprintf("Passing in your own %v is no longer supported", GUIDKey))
```

Comment:
Sorry if I missed a conversation, but my preference is to avoid all `panic`s in this library. Maybe just `maybeLogError`?